pyccel/codegen/printing/fcode.py

```diff
@@ -1099,7 +1099,7 @@ def _print_NumpyRand(self, expr):
         if (not self._additional_code):
             self._additional_code = ''
         var_name = self.scope.get_new_name()
-        var = Variable(expr.dtype, var_name, is_stack_array = all([s.is_constant for s in expr.shape]),
+        var = Variable(expr.dtype, var_name, memory_handling = 'stack',
```

Comment:
These statements don't seem to be equivalent. `memory_handling` should be `heap` if the shape is not known at compile time

Reply:
Yes they are not equivalent, but `allocatable` was always set to False in this case.

Reply:
You are right. There is an error in the case of arrays :+1: